File: web/src/wire/friendship.ts
export type Friendship = {
    uid: string,
    name: string,
    email: string,
    avatarUrl: string,
    directConversationId: string | null,  // null if the friendship hasn't been accepted yet
    status: "pending" | "accepted" | "blocked",
    direction: "incoming" | "outgoing",
}

export type FriendshipDto = {
    uid: string,
    name: string,
    email: string,
    avatar_url: string,
    direct_conversation_id: string | null,
    status: "pending" | "accepted" | "blocked",
    direction: "incoming" | "outgoing",
}


export const toFriendship = (w: FriendshipDto): Friendship => ({
    uid: w.uid,
    name: w.name,
    email: w.email,
    avatarUrl: w.avatar_url,
    directConversationId: w.direct_conversation_id,
    status: w.status,
    direction: w.direction,
}); 

File: wire/friendship.go
package wire

import "github.com/google/uuid"

type Friendship struct {
	UID                  uuid.UUID  `db:"uid" json:"uid"`
	Name                 string     `db:"name" json:"name"`
	Email                string     `db:"email" json:"email"`
	AvatarURL            string     `db:"avatar_url" json:"avatar_url"`
	DirectConversationID *uuid.UUID `db:"direct_conversation_id" json:"direct_conversation_id"`
	Status               string     `db:"status" json:"status"`       // pending | accepted | blocked'
	Direction            string     `db:"direction" json:"direction"` // incoming | outgoing
}
